services/web/server/src/simcore_service_webserver/login/handlers_registration.py

```diff
@@ -249,10 +251,17 @@ async def register(request: web.Request):
 
 
 class RegisterPhoneBody(InputSchema):
-    email: EmailStr
+    email: LowerCaseEmailStr
     phone: str = Field(
         ..., description="Phone number E.164, needed on the deployments with 2FA"
     )
+    username: str = Field(None, description="User name", hidden=True)
+
+    @root_validator(pre=True)
+    @classmethod
+    def _validate_user(cls, values):
+        values["username"] = values["email"].split("@")[0]
```

Comment:
So you try to capture "the original" emails by adding a ``pre``-validator at the root level. That means that the ``email`` you get is actually still NOT validated!


This approach will at least fail with this input:
```python
RegisterBody(email="THIS IS NOT AN EMAIL", ...)
```

Reply:
Yes, you are right, thanks for the catch. In the end, didnt proceed with this idea.